numhask/src/NumHask/Algebra/Abstract/Additive.hs

```diff
@@ -1,78 +1,58 @@
-{-# OPTIONS_GHC -Wall #-}
-{-# LANGUAGE ConstrainedClassMethods #-}
-{-# LANGUAGE ConstraintKinds #-}
 {-# LANGUAGE DeriveFunctor #-}
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
-{-# LANGUAGE GADTs #-}
 {-# LANGUAGE MonoLocalBinds #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE RankNTypes #-}
-{-# LANGUAGE RoleAnnotations #-}
-{-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wall #-}
 
--- | The Group hirarchy
+-- | Additive
 module NumHask.Algebra.Abstract.Additive
-      ( zero
-      , Sum(..)
-      , coerceFA
-      , coerceFA'
-      , coerceTA
-      , coerceTA'
-      , Addition(..)
-      , (+)
-      , (-)
-      , negate
-      , plus
-      )
+  ( Sum(..)
+  , Additive
+  , (+)
+  , (-)
+  , zero
+  , negate
+  , sum
+  )
 where
 
-import qualified Prelude                       as P
-import qualified GHC.Generics                  as P
-import           NumHask.Algebra.Abstract.Group
 import Data.Coerce
-import           Data.Complex                   ( Complex(..) )
-import           Data.Int                       ( Int8
-                                                , Int16
-                                                , Int32
-                                                , Int64
-                                                )
-import           Data.Word                      ( Word
-                                                , Word8
-                                                , Word16
-                                                , Word32
-                                                , Word64
-                                                )
-import           GHC.Natural                    ( Natural(..) )
+import Data.Complex (Complex(..))
+import Data.Int (Int8, Int16, Int32, Int64)
+import Data.Word (Word, Word8, Word16, Word32, Word64)
+import GHC.Natural (Natural(..))
+import NumHask.Algebra.Abstract.Group
+import qualified GHC.Generics as P
+import qualified Prelude as P
 
 newtype Sum a = Sum a
-      deriving (P.Eq, P.Ord, P.Read, P.Show, P.Bounded, P.Generic, P.Generic1, P.Functor)
-
-class (Associative (Sum a), Commutative (Sum a), Unital (Sum a)) => Addition a where
-      sum :: (P.Foldable f) => f a -> a
-      sum = P.foldr (+) zero
-instance (Associative (Sum a), Commutative (Sum a), Unital (Sum a)) => Addition a
+  deriving (P.Eq, P.Ord, P.Read, P.Show, P.Bounded, P.Generic, P.Generic1,
+            P.Functor)
 
-plus :: Magma (Sum a) => a -> a -> a
-plus = coerceFA magma
+class (Associative (Sum a), Commutative (Sum a), Unital (Sum a)) => Additive a
+instance (Associative (Sum a), Commutative (Sum a), Unital (Sum a)) => Additive a
 
 infixl 6 +
-(+) :: Addition a => a -> a -> a
+(+) :: Magma (Sum a) => a -> a -> a
```

Comment:
While I am not against it, I would like to know why this is a good idea. It will open up `+` to all sorts of things, which `base` currently avoids.

I think to place `+` squarely into `num` was a good decision for base. I would like to see at least some laws associated with `+`, otherwise it becomes arbitrary.

Reply:
Having to type the ugly `plus` versus the sublime + has very much been a pain for me. The tests look so much better. Indulge me until someone gets a bug where they use + but didn't realise it didnt mean commutative etc.